sw.js

```diff
@@ -1,41 +1,62 @@
-self.addEventListener('install', function(event) {
+const addResourcesToCache = async (resources) => {
+  const cache = await caches.open("v1");
+  await cache.addAll(resources);
+};
+
+const putInCache = async (request, response) => {
+  const cache = await caches.open("v1");
+  await cache.put(request, response);
+};
```

Comment:
Hm, one option would be to use await twice (which would block the return of responseFromNetwork)

```javascript
const cache = await caches.open('v1');
await cache.put(request, responseFromNetwork.clone()));
return responseFromNetwork;
```

Or just having one await (which would still block the return of responseFromNetwork):

```javascript
const cache = await caches.open('v1');
cache.put(request, responseFromNetwork.clone()));
return responseFromNetwork;
```

Another other option would be to use Promise.then (which I tried to avoid)
```javascript
caches
  .open('v1')
  .then((cache) => cache.put(request, responseFromNetwork.clone()));
return responseFromNetwork;
```

What do you think?

Reply:
Oh I see. Yes, now you've explained it like that I like the original separate-function version :). But I still think you don't need the `await` in line 8, do you? Because nothing is waiting for it.